cmd/shield/commands/misc/curl.go

```diff
@@ -0,0 +1,76 @@
+package misc
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/starkandwayne/shield/api"
+	"github.com/starkandwayne/shield/cmd/shield/commands"
+	"github.com/starkandwayne/shield/cmd/shield/commands/internal"
+	"github.com/starkandwayne/shield/cmd/shield/log"
+)
+
+var Curl = &commands.Command{
+	Summary: "Issue a REST API call and display the output as formatted JSON",
+	Help: &commands.HelpInfo{
+		Flags: []commands.FlagInfo{
+			commands.FlagInfo{
+				Name:   "method",
+				Short:  'm',
+				Valued: true,
+				Desc:   `HTTP request method to use, one of GET, PUT, POST, PATCH, DELETE`,
+			},
+			commands.FlagInfo{
+				Name:       "url",
+				Positional: true,
+				Mandatory:  true,
+				Desc:       `The path component of the URL to curl`,
+			},
+		},
+	},
+	RunFn: func(opts *commands.Options, args ...string) error {
+		log.DEBUG("running 'curl' command")
+
+		var (
+			method string
+			url    string
+			body   string
+			err    error
+		)
+
+		if len(args) == 1 {
+			method = "GET"
+			url = args[0]
+
+		} else if len(args) == 2 {
+			method = args[0]
+			url = args[1]
+
+		} else {
+			return fmt.Errorf("USAGE: shield curl [METHOD] URL\n")
+		}
+
+		if method == "PUT" || method == "PATCH" || method == "POST" {
+			commands.MSG("Please enter your JSON (^D when done)")
+			body, err = internal.ReadAll(os.Stdin)
```

Comment:
Nitpick inc... So, instead of stalling on stdin, we could do this more curl-esque and check if stdin is a terminal first, and only if it isn't (assume its a pipe / redirect), then we read it. I'm not sure if you'd want to interactively write JSON but... up for debate.